test/test_scenario_perception_pipeline.cpp

```diff
@@ -7,92 +7,165 @@
 #include <moveit/moveit_cpp/planning_component.h>
```

Comment:
I think you should separate the declarations form the implementations by adding header files for your classes. That way everything would be more cleaned up and easier to work with. I'd propose this repository structure:

```
launch/
include/
 scenario_perception_pipeline.hpp --> Contains class definitions
src/
  scenario_perception_pipeline.cpp --> Contains all the class definitions
  benchmark_main.cpp --> contains just the main function
test/ --> Usually this contains the unittests (you can neglect it for now until we get to the point where we write unittests)
```

Reply:
I refactored based on your review with a couple of details. Could you take a look at it and then give a feedback again?